src/status_im/transport/inbox.cljs

```diff
@@ -130,119 +92,138 @@
+  [{:keys [db] :as cofx}]
+  (if (and (:account/account db)
+           config/offline-inbox-enabled?)
+    (let [{:keys [peers-summary peers-count]} db]
+      (if (zero? peers-count)
+        (update-mailserver-status :disconnected cofx)
+        (let [wnode (get-current-wnode-address db)
+              mailserver-status (:mailserver-status db)]
+          (if (registered-peer? peers-summary wnode)
+            (handlers-macro/merge-fx cofx
+                                     {::mark-trusted-peer {:web3  (:web3 db)
+                                                           :wnode wnode}}
+                                     (generate-mailserver-symkey))
+            (connect-to-mailserver cofx)))))))
+
+(defn get-topics
+  [db topics discover?]
+  (let [inbox-topics    (:inbox/topics db)
+        discovery-topic (transport.utils/get-topic constants/contact-discovery)
+        topics          (or topics
+                            (map #(:topic %) (vals (:transport/chats db))))]
+    (cond-> (apply conj inbox-topics topics)
+      discover? (conj discovery-topic))))
+
+(defn request-messages [{:keys [topics discover? should-recover?]
+                         :or {should-recover? true
+                              discover?       true}}
+                        {:keys [db] :as cofx}]
+  (let [mailserver-status (:mailserver-status db)
+        sym-key-id        (:inbox/sym-key-id db)
+        wnode             (get-current-wnode-address db)
+        inbox-topics      (get-topics db topics discover?)
+        inbox-ready?      (and (= :connected mailserver-status)
+                               sym-key-id)]
+    (when should-recover?
+      (if inbox-ready?
+        {::request-messages {:wnode      wnode
+                             :topics     (into [] inbox-topics)
+                             :sym-key-id sym-key-id
+                             :web3       (:web3 db)}
+         :db                (assoc db :inbox/fetching? true)
+         :dispatch-later    [{:ms fetching-messages-notification-timeout
+                              :dispatch [:inbox/remove-fetching-notification]}]}
+        {:db (assoc db :inbox/topics inbox-topics)}))))
 
 ;;;; Handlers
 
 (handlers/register-handler-fx
- :inbox/add-peer
- ;; This event adds a wnode to the list of peers
- (fn [_ [_ wnode]]
-   {::add-peer {:wnode wnode}}))
-
-(handlers/register-handler-fx
- :inbox/fetch-peers
- ;; This event fetches the list of peers
- ;; We want it to check if the node has been added
- (fn [_ [_ retries]]
-   {::fetch-peers (or retries 0)}))
-
-(def ^:private ^:const short-retry-delay-ms 300)
-(def ^:private ^:const long-retry-delay-ms 5000)
-(def ^:private ^:const max-retries 10)
-(def ^:private ^:const retries-interval-change-threshold 3)
+ :inbox/mailserver-trusted
+ (fn [{:keys [db] :as cofx} _]
+   (handlers-macro/merge-fx cofx
+                            (update-mailserver-status :connected)
+                            (request-messages {}))))
```

Comment:
not a big deal, but we could add another arity for `request-messages` instead of having `(request-messages {})` twice